fastlane/spec/actions_specs/xcodes_spec.rb

```diff
@@ -0,0 +1,117 @@
+describe Fastlane do
+  describe Fastlane::FastFile do
+    describe "xcodes" do
+      let(:xcode_path) { "/valid/path/to/xcode.app" }
+      let(:xcode_developer_path) { "#{xcode_path}/Contents/Developer/" }
+      let(:xcodes_binary_path) { "/path/to/bin/xcodes" }
+      let(:valid_xcodes_version) { "1.1.0" }
+      let(:outdated_xcodes_version) { "1.0.0" }
+
+      before(:each) do
+        allow(Fastlane::Helper::XcodesHelper).to receive(:find_xcodes_binary_path).and_return(xcodes_binary_path)
+        allow(File).to receive(:exist?).and_call_original
+        allow(File).to receive(:exist?).with(xcodes_binary_path).and_return(true)
+        allow(Fastlane::Actions).to receive(:sh).and_call_original
+        allow(Fastlane::Actions).to receive(:sh).with("#{xcodes_binary_path} installed '14'").and_return(xcode_path)
+        allow(Fastlane::Actions).to receive(:sh).with("#{xcodes_binary_path} version", { log: false }).and_return(valid_xcodes_version)
+      end
+
+      after(:each) do
+        ENV.delete("DEVELOPER_DIR")
+      end
+
+      describe "update_list argument" do
+        it "invokes update command when true" do
+          expect(Fastlane::Actions).to receive(:sh).with(/--update/)
```

Comment:
```suggestion
          expect(Fastlane::Actions).to receive(:sh).with('--update')
```

Reply:
This is intentionally a regex, rspec treat string literals and regexes differently:

<img width="848" alt="image" src="https://user-images.githubusercontent.com/8419048/199477782-6223bba9-7511-4c7d-ba41-867dc5ce8b8a.png">

If I were to use a string, I'd not only have to use the entire exact string, but also expect the shell commands before and after the `/--update/` one 🤷 

This applies to all of the ones you commented on here

Reply:
True, my bad 🤦🏻‍♂️